cmd/agent/app/integrations.go

```diff
@@ -115,6 +120,46 @@ var showCmd = &cobra.Command{
 	RunE:  show,
 }
 
+func detectEnvironment() {
+	rootDir, _ = executable.Folder()
+	for {
+		agentReleaseFile := filepath.Join(rootDir, reqAgentReleaseFile)
+		if _, err := os.Lstat(agentReleaseFile); err == nil {
+			reqAgentReleasePath = agentReleaseFile
+			break
+		}
+
+		parentDir := filepath.Dir(rootDir)
+		if parentDir == rootDir {
+			break
+		}
+
+		rootDir = parentDir
+	}
+
+	pythonMajorVersion = config.Datadog.GetString("python_version")
```

Comment:
the `config` here needs to be set up at the beginning of every command that uses it, with `config.SetupConfig(confFilePath)`, and we need to explicitly handle errors (when the config file isn't present or unreadable). In most cases I guess we could ignore config errors and simply display a message to the user if they happen (?).

I'd appreciate if we split this PR into 2: one for the immediate changes needed for 6.12, and another for the changes we'll need to 6.13.

Reply:
nit: this would be `common.SetupConfig()` from the `cmd/agent/common` package.